File: AppSets-Android/share/src/main/java/xcj/app/share/http/service/AppSetsShareService.kt
package xcj.app.share.http.service

import android.content.Context
import xcj.app.share.http.base.HttpShareDevice
import xcj.app.share.http.model.ContentInfoList
import xcj.app.starter.foundation.http.DesignResponse
import xcj.app.web.webserver.base.ContentDownloadN
import xcj.app.web.webserver.base.FileUploadN

interface AppSetsShareService {
    fun greeting(context: Context, clientHost: String): DesignResponse<String>

    fun ping(context: Context, clientHost: String): DesignResponse<String>

    fun isNeedPin(context: Context, clientHost: String): DesignResponse<Boolean>

    fun pair(
        context: Context,
        clientHost: String,
        pin: Int
    ): DesignResponse<Boolean>

    fun pairResponse(
        context: Context,
        clientHost: String,
        shareToken: String
    ): DesignResponse<Boolean>

    fun prepareSend(
        context: Context,
        clientHost: String,
        shareToken: String,
        uri: String
    ): DesignResponse<Boolean>

    fun prepareSendResponse(
        context: Context,
        clientHost: String,
        shareToken: String,
        isAccept: Boolean,
        preferDownloadSelf: Boolean,
    ): DesignResponse<Boolean>

    fun postText(
        context: Context,
        clientHost: String,
        shareToken: String,
        text: String
    ): DesignResponse<Boolean>

    fun postFile(
        context: Context,
        clientHost: String,
        shareToken: String,
        fileUploadN: FileUploadN
    ): DesignResponse<Boolean>

    fun postFileChunked(
        context: Context,
        clientHost: String,
        shareToken: String,
        fileId: String,
        chunkCount: Int,
        chunk: Int,
        fileUploadN: FileUploadN
    ): DesignResponse<Boolean>

    fun getContent(
        context: Context,
        clientHost: String,
        shareToken: String,
        contentId: String
    ): DesignResponse<ContentDownloadN>

    fun getContentList(
        context: Context,
        clientHost: String,
        shareToken: String,
        uri: String
    ): DesignResponse<ContentInfoList>

    fun exchangeDeviceInfo(
        context: Context,
        clientHost: String,
        device: HttpShareDevice
    ): DesignResponse<HttpShareDevice>
}

File: AppSets-iOS/AppSets/com/app/appsets/share/http/service/AppSetsShareService.swift
//
//  AppSetsShareService.swift
//  AppSets
//
//  Created by caiju Xu on 2025/3/3.
//

protocol AppSetsShareService {
    func greeting(
        context: Context,
        clientHost: String
    ) async -> DesignResponse<String>

    func ping(
        context: Context,
        clientHost: String
    ) async -> DesignResponse<String>

    func isNeedPin(
        context: Context,
        clientHost: String
    ) async -> DesignResponse<Bool>

    func pair(
        context: Context,
        clientHost: String,
        pin: Int
    ) async -> DesignResponse<Bool>

    func pairResponse(
        context: Context,
        clientHost: String,
        shareToken: String
    ) async -> DesignResponse<Bool>

    func prepareSend(
        context: Context,
        clientHost: String,
        shareToken: String,
        uri: String
    ) async -> DesignResponse<Bool>

    func prepareSendResponse(
        context: Context,
        clientHost: String,
        shareToken: String,
        isAccept: Bool,
        preferDownloadSelf: Bool
    ) async -> DesignResponse<Bool>

    func postText(
        context: Context,
        clientHost: String,
        shareToken: String,
        text: String
    ) async -> DesignResponse<Bool>

    func postFile(
        context: Context,
        clientHost: String,
        shareToken: String,
        fileUploadN: FileUploadN
    ) async -> DesignResponse<Bool>

    func postFileChunked(
        context: Context,
        clientHost: String,
        shareToken: String,
        fileId: String,
        chunkCount: Int,
        chunk: Int,
        fileUploadN: FileUploadN
    ) async -> DesignResponse<Bool>

    func getContent(
        context: Context,
        clientHost: String,
        shareToken: String,
        contentId: String
    ) async -> DesignResponse<ContentDownloadN>

    func getContentList(
        context: Context,
        clientHost: String,
        shareToken: String,
        uri: String
    ) async -> DesignResponse<ContentInfoList>

    func exchangeDeviceInfo(
        context: Context,
        clientHost: String,
        device: HttpShareDevice
    ) async -> DesignResponse<HttpShareDevice>
}
